src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -335,8 +336,32 @@ static ss::future<response_ptr> do_handle(alter_op_context& octx) {
         return octx.rctx.respond(std::move(octx.response));
     }
 
-    if (!octx.rctx.authorized(
-          security::acl_operation::alter, security::default_cluster_name)) {
+    auto authz = octx.rctx.authorized(
+      security::acl_operation::alter, security::default_cluster_name);
+
+    const auto additional_resources = [&octx]() {
+        std::vector<model::topic> topics;
+        topics.reserve(octx.request.data.topics.size());
+        std::transform(
+          octx.request.data.topics.begin(),
+          octx.request.data.topics.end(),
+          std::back_inserter(topics),
+          [](const reassignable_topic& t) { return t.name; });
+
+        return topics;
+    };
+
+    if (!octx.rctx.audit(
+          get_audit_event_type<decltype(octx.request)>,
+          additional_resources())) {
```

Comment:
concerned about this `additiona_resources()` parameters here. seems like it gets built even if it is never used (e.g. auditing is disabled).

Reply:
yeah @graphcareful made a  similar remark about generating the message before chekcing that that event type was enabled.  I have an idea on how to solve this issue.